presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestRemoveIdentityCasts.java

```diff
@@ -0,0 +1,127 @@
+                .build());
+
+        queryRunner.createCatalog(queryRunner.getDefaultSession().getCatalog().get(),
+                new TpchConnectorFactory(queryRunner.getNodeManager(), 1),
+                ImmutableMap.<String, String>of());
+    }
+
+    @Test
+    public void testRemoveIdentityCastUnparameterizedColumnData()
+    {
+        @Language("SQL") String sql = "select cast(nationkey as bigint) from nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertNotPlan(sql, pattern);
+    }
+
+    @Test
+    public void testKeepNonIdentityCastUnparameterizedColumnData()
+    {
+        @Language("SQL") String sql = "select cast(nationkey as smallint) from nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertPlan(sql, pattern);
+    }
+
+    @Test
+    public void testRemoveIdentityCastParameterizedColumnData()
+    {
+        @Language("SQL") String sql = "select cast(name as varchar(25)) from nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertNotPlan(sql, pattern);
+    }
+
+    @Test
+    public void testKeepNonIdentityCastParameterizedColumnData()
+    {
+        @Language("SQL") String sql = "select cast(name as varchar(30)) from nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertPlan(sql, pattern);
+    }
+
+    @Test
+    public void testKeepNonIdentityCastDefaultParameterizedColumnData()
+    {
+        @Language("SQL") String sql = "SELECT CAST(name AS varchar) FROM nation";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertPlan(sql, pattern);
+    }
+
+    @Test
+    public void testRemoveIdentityCastImplicitBigintData()
+    {
+        @Language("SQL") String sql = "select cast(bigint '5' as bigint)";
+        PlanMatchPattern pattern = anyTree(
+                project(anyTree()).withAssignment("Cast")
+        );
+        assertNotPlan(sql, pattern);
+    }
+
+    private void assertNotPlan(@Language("SQL") String sql, PlanMatchPattern pattern)
+    {
+        Plan actualPlan = plan(sql);
+        queryRunner.inTransaction(transactionSession -> {
+            PlanAssert.assertNotPlan(transactionSession, queryRunner.getMetadata(), actualPlan, pattern);
+            return null;
+        });
+    }
+
+    private void assertPlan(@Language("SQL") String sql, PlanMatchPattern pattern)
+    {
+        Plan actualPlan = plan(sql);
+        queryRunner.inTransaction(transactionSession -> {
+            PlanAssert.assertPlan(transactionSession, queryRunner.getMetadata(), actualPlan, pattern);
+            return null;
+        });
+    }
+
+    private Plan plan(@Language("SQL") String sql)
```

Comment:
I it is already defined in several classes. Maybe you could extract an abstract class which defines this methods like `plan`, `assertPlan` etc. And reuse it in other test class which asserts logical plan tree.